File: spring-web-java/src/main/java/spring/web/java/domain/converter/CryptoAttributeConverter.java
package spring.web.java.domain.converter;

import org.springframework.stereotype.Component;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;
import lombok.RequiredArgsConstructor;
import spring.web.java.infrastructure.util.crypto.CryptoUtil;

@Component
@Converter
@RequiredArgsConstructor
public class CryptoAttributeConverter implements AttributeConverter<String, String> {

	private final CryptoUtil hexAESCryptoUtil;

	@Override
	public String convertToDatabaseColumn(String s) {
		return hexAESCryptoUtil.encrypt(s);
	}

	@Override
	public String convertToEntityAttribute(String s) {
		return hexAESCryptoUtil.decrypt(s);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/domain/converter/CryptoAttributeConverter.kt
package spring.web.kotlin.domain.converter

import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter
import org.springframework.stereotype.Component
import spring.web.kotlin.infrastructure.util.crypto.CryptoUtil

@Component
@Converter
class CryptoAttributeConverter(
    private val hexAESCryptoUtil: CryptoUtil
) : AttributeConverter<String?, String?> {
    override fun convertToDatabaseColumn(attribute: String?) = attribute?.let { hexAESCryptoUtil.encrypt(it) }
    override fun convertToEntityAttribute(dbData: String?) = dbData?.let { hexAESCryptoUtil.decrypt(it) }
}
